### ISSUE TITLE
use modern cmake

### ISSUE THREAD

**Issue by prudhomm** [berenger-eu/specx#4]
this is not a good idea to use include_directories, it is preferred to use target_include_directories which way cleaner.
it is called modern cmake: see eg here https://cliutils.gitlab.io/modern-cmake/chapters/intro/dodonot.html

I can give a shot at fixing this if you like  @berenger-eu

**Comment by berenger-eu** [comment]
Yes I know, it is on my list since years.... I will try to do it (start now and hope to achieve in the coming weeks).
